### ISSUE TITLE
Design for render quality configuration

### ISSUE THREAD

**Issue by pauldendulk** [Mapsui/Mapsui#2213]
**The problem.**
SkiaSharp has options to improve render quality but this impacts performance. Examples are the use of IsAntialias or FilterQuality. In some cases you want to the best quality, in others you want the best performance. This issue discusses a way to make this configurable.

**The solution**
Not decided.

**Comment by pauldendulk** [comment]
Some thoughts
- Different situations require different quality and performance:
   - Widgets will be few in number so can always be high quality.
  - MyLocation layer has just one feature and should be high quality.
  - Other layers could be high or low quality depending on the number of features. 
- These configurations apply specifically to SkiaSharp. We do not want SkiaSharp specific logic to be in the MapControl, Map or Layers.
- An option would be to use abstract terms to describe quality level (high, medium, low) that could be interpreted on every platform but I don't like that much because you do not know what is going on without looking at the code and it might not be sufficient for every renderer. You really want to know what is exactly happening. 
- **Instructions as string?**: Perhaps we can have a list of string with quality instruction where different strings can be interpreted for specific cases. For example:
   `ImageLayer.Quality.Add("FilterQualityHigh");`
   `Layer.Quality.Add("IsAntiAliasTrue");`
- **Style quality**: In the example above we configure the layer, but a layer can contains different types of features. Usually a quality instruction applies specifically to a Style and a StyleRenderer. So perhaps the quality instructions should be on the Style to instruct it's corresponding renderer. For example, the RasterStyle would be the place to configure the FilterQuality.(High|Medium|Low|None) setting.
  `rasterStyle.Quality.Add("FilterQualityHigh");`
- Perhaps it should not be a list but a dictionary, so the rasterStyleRenderer could specifically check if there is a key for 'FilterQuality'. For IsAntialias it makes less sense, since either True or False would be the default.

**Comment by charlenni** [comment]
It isn't correct, that a Widget (I still prefer Overlay for this) has always a few things to draw. They could get very complicated.

For me, it is a matter of the Layer or, better, the Style. I think, it is enough to have a property for setting the quality, which could have the settings "Normal" and "High". That should be enough.

**Comment by pauldendulk** [comment]
> It isn't correct, that a Widget (I still prefer Overlay for this) has always a few things to draw. They could get very complicated.

Good point. We may want to tread widgets the same with. The idea of using styles cannot be applied to widgets though.
 
> For me, it is a matter of the Layer or, better, the Style. I think, it is enough to have a property for setting the quality, which could have the settings "Normal" and "High". That should be enough.

I think we need to make a lot of choices if we do it like that. We have to think about the choices and sometimes they may not be good enough. In the FilterQuality there are four options and currently we use neither the highest or the lowest.

I still like the idea of adding this option to the IStyle

Not sure if the Quality tag should be a single string or a key value, or a string split by some separator. A very simple approach would be to literally use the name and value of the skia property.
